list2/ex29/index.js

```diff
@@ -0,0 +1,85 @@
+import { generateMatrix, printMatrix } from "../helper.js";
+
+const sumOf = {
+  fourthRow: 0,
+  secondColumn: 0,
+  mainDiagonal: 0,
+  allElements: 0,
+};
+
+function sumOfRow({ matrix, columnCount, rowIndex }) {
+  for (let columnIndex = 0; columnIndex < columnCount; columnIndex++) {
+    sumOf.fourthRow += matrix[rowIndex][columnIndex];
+  }
+  return sumOf.fourthRow;
+}
+
+function sumOfColumn({ matrix, rowCount, columnIndex }) {
+  for (let rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+    sumOf.secondColumn += matrix[rowIndex][columnIndex];
+  }
+  return sumOf.secondColumn;
+}
+
+function sumOfDiagonal({ matrix, rowCount, columnCount }) {
+  for (let columnIndex = 0; columnIndex < columnCount; columnIndex++) {
+    for (let rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+      if (columnIndex === rowIndex) {
+        sumOf.mainDiagonal += matrix[rowIndex][columnIndex];
+      }
+    }
+  }
+  return sumOf.mainDiagonal;
+}
+
+function sumOfAllMatrix({ matrix, rowCount, columnCount }) {
+  for (let columnIndex = 0; columnIndex < columnCount; columnIndex++) {
+    for (let rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+      sumOf.allElements += matrix[rowIndex][columnIndex];
+    }
+  }
+  return sumOf.allElements;
+}
```

Comment:
You don't need the `const sumOf = {` const. I'd suggest declaring a function scoped variables and return at the end.